### DISCUSSION TITLE
Creating Dynamic Form

### DISCUSSION THREAD

**Discussion by imadeelnazar** [react-hook-form/react-hook-form#9648] [Q&A]
Hi @bluebill1049 Would love to know your input or any solution you suggest.

Creating Dynamic Form values got submitted successfully and Returning Fine, What i want is

**=>Edit (Data which is coming from DB)
=>Delete
=>Append**

i have tried useForm passing defaultvalues to it but still no success.

Here is the URL to Sandbox 
https://codesandbox.io/s/youthful-merkle-s807v5

Thank You

**Comment by Moshyfawn** [comment]
To set `defaultValues` that come from an async storage _(like your Backend, for example)_, you can create an effect like this
```js
const asyncData = useAsyncData()

const { reset } = useForm({
  defaultValues: {
    test: []
  }
})

useEffect(() => {
  if (asyncData) {
    reset(asyncData)
  }
}, [reset, asyncData])
```

Another way would be to utilize the new `values` prop from the latest [v7.41.0 release](https://github.com/react-hook-form/react-hook-form/releases/tag/v7.41.0)
